app/src/test/java/ch/hikemate/app/model/elevation/ElevationRepositoryTest.kt

```diff
@@ -0,0 +1,253 @@
+          LatLong(35.0, 30.0),
+          LatLong(45.0, -5.0),
+          LatLong(50.0, 50.0),
+          LatLong(60.5, 60.5),
+          LatLong(40.0, -70.0),
+          LatLong(50.0, 90.0),
+          LatLong(55.0, 100.0),
+          LatLong(30.0, -120.0),
+      )
+
+  private val longResponse =
+      Response.Builder()
+          .code(200)
+          .message("OK")
+          .body(
+              """
+{
+   "elevations":
+   [
+      515.0,
+      545.0,
+      117.0,
+      620.0,
+      480.0,
+      200.0,
+      750.0,
+      1020.0,
+      890.0,
+      1500.0,
+      340.0,
+      670.0
+   ]
+}
+            """
+                  .trimIndent()
+                  .replace("\n", "")
+                  .toResponseBody("application/json; charset=utf-8".toMediaType()))
+          .protocol(Protocol.HTTP_1_1)
+          .header("Content-Type", "application/json")
+          .request(mockk())
+          .build()
+
+  private val simpleResponse =
+      Response.Builder()
+          .code(200)
+          .message("OK")
+          .body(
+              """
+{
+   "elevations":
+   [
+      515.0,
+      545.0,
+      117.0
+   ]
+}
+            """
+                  .trimIndent()
+                  .replace("\n", "")
+                  .toResponseBody("application/json; charset=utf-8".toMediaType()))
+          .protocol(Protocol.HTTP_1_1)
+          .header("Content-Type", "application/json")
+          .request(mockk())
+          .build()
+
+  private val requestEntityTooLargeResponse =
+      Response.Builder()
+          .code(413)
+          .message("Request Entity Too Large")
+          .body("Request Entity Too Large".toResponseBody(null))
+          .protocol(Protocol.HTTP_1_1)
+          .header("Content-Type", "text/plain")
+          .request(mockk())
+          .build()
+
+  private val serverErrorResponse =
+      Response.Builder()
+          .code(500)
+          .message("Internal Server Error")
+          .body("Internal Server Error".toResponseBody(null))
+          .protocol(Protocol.HTTP_1_1)
+          .header("Content-Type", "text/plain")
+          .request(mockk())
+          .build()
+
+  private val dispatcher = UnconfinedTestDispatcher()
+
+  @Before
+  fun setup() {
+    Dispatchers.setMain(dispatcher)
+    client = mockk()
+    elevationRepository = ElevationServiceRepository(client, dispatcher)
+  }
+
+  @Test
+  fun worksWithSimpleAnswer() = runTest {
+    val call: Call = mockk()
+    coEvery { client.newCall(any()) } returns call
+
+    coEvery { call.enqueue(any()) } answers
+        {
+          firstArg<Callback>().onResponse(call, simpleResponse)
+        }
+
+    var onSuccessCalled = false
+
+    elevationRepository.getElevation(
+        latLongList,
+        { list ->
+          assertEquals(smallList, list)
+          onSuccessCalled = true
+        }) {
+          fail("Failed to fetch routes from Overpass API")
+        }
+
+    // Wait for the coroutine to finish
+    advanceUntilIdle()
```

Comment:
Question: is this requried? Have you tried to use `UnconfinedDispatcher` everywhere instead with `runTest(dispatcher)`?

Reply:
I tried it myself locally, does not work. Forget what I said.

Reply:
Thanks x)